core/src/main/java/com/linecorp/armeria/client/encoding/ZlibStreamDecoder.java

```diff
@@ -40,6 +41,7 @@
 
     @Override
     public HttpData decode(HttpData obj) {
+        obj = PooledObjects.toUnpooled(obj);
```

Comment:
How about writing directly if it's bytebuf?

Reply:
That's a good idea. Fixed!